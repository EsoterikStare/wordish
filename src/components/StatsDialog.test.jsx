import { screen } from '@testing-library/react';
import React from 'react';

import StatsDialog from './StatsDialog';

import {
  loseGameState,
  mockLocalStorage,
  newGameState,
  setup,
  winGameState,
} from '../test-utils';

const noop = () => {};

describe('StatsDialog', () => {
  it('should be defined', () => {
    expect(StatsDialog).toBeDefined();
  });
  it('should render when open is true', () => {
    setup(<StatsDialog onClose={noop} open />);
    expect(screen.getByRole('dialog')).toBeInTheDocument();
  });
  it('should fire provided onClose when closed', async () => {
    const onCloseSpy = jest.fn();
    const { user } = setup(<StatsDialog onClose={onCloseSpy} open />);
    await user.keyboard('{Escape}');

    expect(onCloseSpy).toHaveBeenCalledTimes(1);
  });
  describe('When gameState is playing', () => {
    beforeEach(() => {
      mockLocalStorage(newGameState);
    });
    it('should display the correct heading', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByRole('heading', { name: /Don't give up!/i })).toBeInTheDocument();
    });
    it('should have one button', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getAllByRole('button')).toHaveLength(1);
    });
    it('should have a "new word" button', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByRole('button', { name: /new word/i })).toBeInTheDocument();
    });
    it('should look as expected', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByRole('dialog')).toMatchSnapshot();
    });
  });
  describe('When gameState is lose', () => {
    beforeEach(() => {
      mockLocalStorage(loseGameState);
    });
    it('should display the correct heading', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByRole('heading', { name: /That's ok! Try another!/i })).toBeInTheDocument();
    });
    it('should give the correct solution', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByText(/The word was \w{5,6}/i)).toBeInTheDocument();
    });
    it('should have one button', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getAllByRole('button')).toHaveLength(1);
    });
    it('should have an "another!" button', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByRole('button', { name: /another!/i })).toBeInTheDocument();
    });
    it('should look as expected', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByRole('dialog')).toMatchSnapshot();
    });
  });
  describe('When gameState is win', () => {
    beforeEach(() => {
      mockLocalStorage(winGameState);
    });
    it('should display the correct heading', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByRole('heading', { name: /You did it!/i })).toBeInTheDocument();
    });
    it('should have two buttons', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getAllByRole('button')).toHaveLength(2);
    });
    it('should have a "share" button', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByRole('button', { name: /share/i })).toBeInTheDocument();
    });
    it('should have an "another!" button', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByRole('button', { name: /another!/i })).toBeInTheDocument();
    });
    it('should look as expected', () => {
      setup(<StatsDialog onClose={noop} open />);
      expect(screen.getByRole('dialog')).toMatchSnapshot();
    });
  });
});
