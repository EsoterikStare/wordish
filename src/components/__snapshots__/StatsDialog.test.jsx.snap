// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsDialog When gameState is lose should look as expected 1`] = `
<div
  aria-describedby="stats-dialog-content-text"
  aria-labelledby="stats-dialog-title"
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1qxadfk-MuiPaper-root-MuiDialog-paper"
  role="dialog"
>
  <h2
    class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
    data-testid="stats-dialog-title"
    id="stats-dialog-title"
  >
    That's ok! Try another!
  </h2>
  <div
    class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
  >
    <p
      class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-eah5ga-MuiTypography-root-MuiDialogContentText-root"
    >
      The word was ABOUND
    </p>
    <p
      class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-c0dc4c-MuiTypography-root-MuiDialogContentText-root"
      data-testid="stats-dialog-content-text"
    >
      Someday, I'll have some metrics to show you. But not today. =(
    </p>
  </div>
  <div
    class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-jcwuwq-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Another!
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`StatsDialog When gameState is playing should look as expected 1`] = `
<div
  aria-describedby="stats-dialog-content-text"
  aria-labelledby="stats-dialog-title"
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1qxadfk-MuiPaper-root-MuiDialog-paper"
  role="dialog"
>
  <h2
    class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
    data-testid="stats-dialog-title"
    id="stats-dialog-title"
  >
    Don't give up!
  </h2>
  <div
    class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
  >
    <p
      class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-c0dc4c-MuiTypography-root-MuiDialogContentText-root"
      data-testid="stats-dialog-content-text"
    >
      Someday, I'll have some metrics to show you. But not today. =(
    </p>
  </div>
  <div
    class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-etq55k-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      New word
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`StatsDialog When gameState is win should look as expected 1`] = `
<div
  aria-describedby="stats-dialog-content-text"
  aria-labelledby="stats-dialog-title"
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm css-1qxadfk-MuiPaper-root-MuiDialog-paper"
  role="dialog"
>
  <h2
    class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
    data-testid="stats-dialog-title"
    id="stats-dialog-title"
  >
    You did it!
  </h2>
  <div
    class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
  >
    <p
      class="MuiTypography-root MuiDialogContentText-root MuiTypography-body1 MuiDialogContentText-root css-c0dc4c-MuiTypography-root-MuiDialogContentText-root"
      data-testid="stats-dialog-content-text"
    >
      Someday, I'll have some metrics to show you. But not today. =(
    </p>
  </div>
  <div
    class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-1sw5mi1-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Share
      <span
        class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
      >
        <svg
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="18"
            cy="5"
            r="3"
          />
          <circle
            cx="6"
            cy="12"
            r="3"
          />
          <circle
            cx="18"
            cy="19"
            r="3"
          />
          <line
            x1="8.59"
            x2="15.42"
            y1="13.51"
            y2="17.49"
          />
          <line
            x1="15.41"
            x2="8.59"
            y1="6.51"
            y2="10.49"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-jcwuwq-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Another!
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
